actors/market/tests/publish_storage_deals_failures.rs

```diff
@@ -203,4 +208,268 @@ mod publish_storage_deals_failures {
         };
         assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
     }
+
+    #[test]
+    fn fail_when_client_has_some_funds_but_not_enough_for_a_deal() {
+        let mut rt = setup();
+
+        //
+        add_participant_funds(&mut rt, CLIENT_ADDR, TokenAmount::from(100u8));
```

Comment:
This is missing from the go test but we should include an assertion that this amount of funds is strictly lower than the deals total cost.  To do this set client funds to a variable and after proposal generation assert that it is less than deal1.client_balance_requirements.